src/System.Drawing.Common/src/System/Drawing/FontFamily.cs

```diff
@@ -236,14 +160,8 @@ private void Dispose(bool disposing)
                     Debug.Assert(status == SafeNativeMethods.Gdip.Ok, "GDI+ returned an error status: " + status.ToString(CultureInfo.InvariantCulture));
 #endif
                 }
-                catch (Exception ex)
+                catch (Exception ex) when (!ClientUtils.IsCriticalException(ex))
```

Comment:
For all intents and purposes it's the same.  There's some sublety around stack unwinding, but for our purposes the change is fine.

Presumably the Debug.Fail was removed because this is actually catching exceptions in our tests?